src/docker-in-docker/install.sh

```diff
@@ -325,7 +326,8 @@ tee /usr/local/share/docker-init.sh > /dev/null \
 
 set -e
 
-AZURE_DNS_AUTO_DETECTION=$AZURE_DNS_AUTO_DETECTION
+export AZURE_DNS_AUTO_DETECTION=$AZURE_DNS_AUTO_DETECTION
+export DOCKER_DEFAULT_ADDRESS_POOL=$DOCKER_DEFAULT_ADDRESS_POOL
```

Comment:
These don't need to be exported as they are only being used inside the script they are being injected into.

```suggestion
AZURE_DNS_AUTO_DETECTION=$AZURE_DNS_AUTO_DETECTION
DOCKER_DEFAULT_ADDRESS_POOL=$DOCKER_DEFAULT_ADDRESS_POOL
```

Reply:
I'd also wrap these in `{ }` to stay consistent with the rest of the file (or at least the places touched recently).

Reply:
If we don't export these, when `sudo -E /bin/sh -c "${dockerd_start}"` is run these variables are not available.

It does have the variables when run with eval, though.

Reply:
I was able to avoid exporting and using `sudo -E` by defining the variables inside `dockerd_start` directly in the generated `docker-init.sh` script. That way they're defined automatically when the script in that variable is run using `sudo`.